core-java-modules/core-java-concurrency-advanced/src/test/java/com/baeldung/concurrent/interrupt/InterruptExampleUnitTest.java

```diff
@@ -0,0 +1,34 @@
+package com.baeldung.concurrent.interrupt;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class InterruptExampleUnitTest {
+
+    @Test
+    public void whenPropagateException_thenAssertionSucceeds() {
```

Comment:
```suggestion
    public void whenPropagateException_thenThrowsInterruptedException() {
```

Reply:
We can be more explicit about the behaviors observed in the 'then' clause of the test method name.